protocol/src/codec.rs

```diff
@@ -56,6 +62,18 @@ impl Decoder for MessageCodec {
     }
 }
 
+fn validate_payload_length(length: u64) -> Result<()> {
+    if length > MAX_MESSAGE_SIZE as u64 {
+        bail!(
```

Comment:
Could you just use an `Err` variant here? There's no need for bail I don't think?